_posts/2021-07-13-gs_00003_terminal_basics2.md

```diff
@@ -0,0 +1,394 @@
+firefox with CTRL-C and run it, again, with a tiny change:
+
+    $ firefox &
+
+The "&" indicates that firefox shall run in the background. You will be prompted
+a number, which is the so-called **process id**, and you are able to enter
+further commands. If you want to stop firefox from the command line, you can
+kill it using one of the following:
+
+    $ kill <process id>
+    $ killall firefox
+
+The process id is the one you were shown when starting firefox. The "kill"
+command will, therefore, only kill that specific firefox process. The "killall"
+command will kill all firefox instances. You might wonder if you really need to
+take note of the process id, and the answer is, of course, no! Always remember
+that programmers are lazy! You can use the "ps" command to find the process id
+(PID).
+
+    $ ps
+
+There are many options you can add to the command. Check the man page for a full
+list. An often used command is
+
+    $ ps aux
+
+which gives a list of all running processes. Since that list is usually long,
+you can search for a specific process:
+
+    $ ps aux | grep firefox
+
+You should be able to find your PID and also might see that firefox created
+process children. If you kill the main process, you will kill all children as
+well. Damn ... This world is so cruel! Sometimes it is also nice to use:
+
+    $ ps axjf
+
+This will show you a process tree. You can see that firefox is a child of
+something called "bash". We will talk about it in the next post, but it means
+your terminal. Again, you can see the children of firefox.
+
+One last thing! Sometimes, it is useful to monitor the CPU utilization and see
+which process uses it. A nice tool to do so is:
+
+    $ htop
+
+Our computer uses right now 814 MB of RAM and has a CPU utilization of around
+20%. What about yours? If you have multiple cores, you can see the utilization
+per core. Exit "htop" using CTRL-C.
+
+<div class='note-example' markdown="1">
+Real-life Example:
+
+You notice that your computer is suddenly very slow. Especially in software
+development, a process might use a huge amount of computing power due to a bug.
+This can have a slowdown of your computer as consequence. There could also be a
+process in the background that did not close as expected. With "htop", you can
+identify such processes and kill them.
+
+</div>
+
+# Summary
+
+In the last two posts you have learned in total 17 commands:
+
+    pwd, ls, cd, touch, mkdir, cp, mv, rm, cat, nano, find, grep, pacman, kill, killall, ps, htop
+
+If you compare that with the number of vocabularies you have to learn for a
+foreign language, it is really not much. To be fair, there are many parameters
+you can use with the commands, but still, it should just take a little practice
+to learn them. And if you forgot something, there is still the man page where
+you can look it up. Of course, there are dozens of other commands. But with
+these, you come already pretty far!
```

Comment:
Sounds too "german". Maybe like:

But with these, you can already accomplish most every day tasks in Linux!

Reply:
to be changed